### PR TITLE
elasticsearchPlugins: fix plugins, add s3-repository and gcs-repository plugin

### PR DESCRIPTION
**Pull request by offlinehacker**
###### Motivation for this change

Fixing of elasticsearch plugins and s3 and gcs repo integration

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/servers/search/elasticsearch/plugins.nix [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
+  esVersion = elasticsearch.version;
+
   esPlugin = a@{
     pluginName,
     installPhase ? ''
       mkdir -p $out/config
       mkdir -p $out/plugins
-      ES_HOME=$out ${elasticsearch-oss}/bin/elasticsearch-plugin install --batch -v file://$src
+      ln -s ${elasticsearch}/lib $out/lib
+      ES_HOME=$out ${elasticsearch}/bin/elasticsearch-plugin install --batch -v file://$src
```
Comment (Mic92) [inline pkgs/servers/search/elasticsearch/plugins.nix]: I suppose this marks every plugin as unfree?

Comment (offlinehacker) [reply pkgs/servers/search/elasticsearch/plugins.nix]: Ahh nvm i forget to commit changed `all-packages.nix`

Comment (offlinehacker) [reply pkgs/servers/search/elasticsearch/plugins.nix]: Ok, fixed now

**Comment by offlinehacker** [comment]
@GrahamcOfBorg eval

**Comment by offlinehacker** [comment]
I'm merging this as it fixes elasticsearch plugins.

**Comment by offlinehacker** [comment]
@GrahamcOfBorg build elasticsearchPlugins elasticsearchPlugins7

**Pull request merged**